ui/imports/utils/Constants.qml

```diff
@@ -466,7 +466,28 @@ QtObject {
         readonly property int blockedContacts: 6
     }
 
+    readonly property QtObject keypair: QtObject {
+        readonly property int nameLengthMax: 20
+        readonly property int nameLengthMin: 5
+    }
+
     readonly property QtObject validators: QtObject {
+        readonly property list<StatusValidator> keypairName: [
+            StatusValidator {
+                name: "startsWithSpaceValidator"
+                validate: function (t) { return !t.startsWith(" ") }
+                errorMessage: qsTr("Keypair starting with whitespace are not allowed")
+            },
+            StatusRegularExpressionValidator {
+                regularExpression: /^[a-zA-Z0-9\-_ ]+$/
```

Comment:
`Constants.regularExpressions.alphanumericalExpanded`?

Reply:
`alphanumericalExpanded` is this `/^$|^[a-zA-Z0-9\-_ ]+$/` which allows for an empty string, that's why I didn't use it.

Reply:
Ah right, I see